yt_dlp/extractor/lastfm.py

```diff
@@ -4,51 +4,27 @@
 import re
 
 from .common import InfoExtractor
-from ..utils import int_or_none
+from ..utils import int_or_none, format_field
 
 
 class LastFMPlaylistBaseIE(InfoExtractor):
-    def _extract_current_page(self, url):
-        return int_or_none(self._search_regex(
-            r'\bpage=(\d+)', url, 'page', default=None))
-
-    def _extract_last_page(self, webpage):
-        last_page = None
-        pagination_list = self._search_regex(
-            r'(?s)class="[^"]*pagination-list[^"]*"[^>]*>(.+?)</ul>', webpage, 'pagination_list', fatal=False,
-            default=None)
-
-        if pagination_list:
-            last_page = max(map(int_or_none, re.findall(r'(\d+)', pagination_list)))
-
-        return last_page
-
-    def _extract_entries(self, webpage):
-        tbody = self._search_regex(
-            r'(?s)tbody\s+data-playlisting-add-entries[^>]*>(.+?)</tbody>', webpage, 'tbody')
-        return [
-            self.url_result(player_url, 'Youtube')
-            for player_url in re.findall(r'href="([^"]+youtube.com/watch[^"]+)"', tbody)
-        ]
-
-    def _download_page(self, url, current_page_number, last_page_number, playlist_id):
-        note = 'Downloading page'
-        query = {}
-
-        if last_page_number:
-            note = '%s %d of %d' % (note, current_page_number, last_page_number)
-            query['page'] = current_page_number
-
-        return self._download_webpage(url, playlist_id, note, query=query)
-
     def _entries(self, url, playlist_id):
         webpage = self._download_webpage(url, playlist_id)
-        last_page_number = self._extract_last_page(webpage)
-        start_page_number = self._extract_current_page(url) or 1
+        start_page_number = int_or_none(self._search_regex(
+            r'\bpage=(\d+)', url, 'page', default=None)) or 1
+
+        last_page_number = int_or_none(self._search_regex(
+            r'>(\d+)</a>[^<]*</li>[^<]*<li[^>]+class="pagination-next', webpage, 'last_page', default=None))
```

Comment:
```suggestion
        start_page_number = int(self._search_regex(
            r'\bpage=(\d+)', url, 'page', default=None)) or 1

        last_page_number = int(self._search_regex(
            r'>(\d+)</a>[^<]*</li>[^<]*<li[^>]+class="pagination-next', webpage, 'last_page', default=None))
```
since it's matched with `/d+`